File: 590 - N-ary Tree Postrorder Traversal.java
import java.util.*;

//import org.w3c.dom.Node;

//import com.sun.org.apache.bcel.internal.classfile.Node;

/*
//Definition for a Node.
class Node {
 public int val;
 public List<Node> children;

 public Node() {}

 public Node(int _val) {
     val = _val;
 }

 public Node(int _val, List<Node> _children) {
     val = _val;
     children = _children;
 }
};
*/

class Solution {
    public List<Integer> postorder(Node root) {
        Node t;
        if (root == null) {
            return new ArrayList<Integer>();
        }
        Deque<Node> s = new ArrayDeque<>();
        List<Integer> res = new ArrayList<>();
        s.push(root);
        while (s.size() > 0) {
            t = s.pop();
            res.add(t.val);
            for (Node child : t.children) {
                s.push(child);
            }
        }
        Collections.reverse(res);
        return res;
    }
}

File: 590 - N-ary Tree Postrorder Traversal.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution:
    res = []
    def postorder(self, root: 'Node') -> List[int]:
        self.res.clear()
        self.dfs(root)
        return self.res
    def dfs(self, root):
        if root is None:
            return
        for child in root.children:
            self.dfs(child)
        self.res.append(root.val)
class Solution:
    res = []
    def postorder(self, root: 'Node') -> List[int]:
        self.res.clear()
        self.dfs(root)
        return self.res[::-1]
    def dfs(self, root):
        if root is None:
            return
        s= [root]
        while len(s):
            t = s.pop()
            self.res.append(t.val)
            for child in t.children:
                s.append(child)
        
            